src/api/app/models/bs_request/find_for/project.rb

```diff
@@ -2,45 +2,56 @@ class BsRequest
   module FindFor
     class Project < Base
       def all
-        inner_or = []
-        @relation, inner_or = extend_relation('source', @relation, roles, package_name, subprojects, project_name, inner_or)
-        @relation, inner_or = extend_relation('target', @relation, roles, package_name, subprojects, project_name, inner_or)
-
-        if (roles.empty? || roles.include?('reviewer')) &&
-           (states.empty? || states.include?('review'))
-          @relation = @relation.references(:reviews)
-          review_states.each do |review_state|
-            @relation = @relation.includes(:reviews)
-            inner_or << if package_name.blank?
-                          "(reviews.state=#{quote(review_state)} and reviews.by_project=#{quote(project_name)})"
-                        else
-                          "(reviews.state=#{quote(review_state)} and reviews.by_project=#{quote(project_name)} and reviews.by_package=#{quote(package_name)})"
-                        end
-          end
-        end
-        if inner_or.empty?
-          @relation.none
-        else
-          @relation.where(inner_or.join(' or '))
-        end
+        return @relation unless project_name.present? || package_name.present?
+
+        @where_conditions = []
+        @where_values = []
+
+        bs_request_actions_conditions('source')
+        bs_request_actions_conditions('target')
+
+        reviews_conditions
+
+        @relation = @relation.where("(#{@where_conditions.join(') or (')})", *@where_values.flatten) if @where_conditions.present?
       end
 
       private
 
-      def extend_relation(source_or_target, requests, roles, package, subprojects, project, inner_or)
-        if roles.empty? || roles.include?(source_or_target)
-          inner_or << if package.blank?
-                        if subprojects.blank?
-                          "bs_request_actions.#{source_or_target}_project=#{quote(project)}"
-                        else
-                          "(bs_request_actions.#{source_or_target}_project like #{quote("#{project}:%")})"
-                        end
-                      else
-                        "(bs_request_actions.#{source_or_target}_project=#{quote(project)} and " \
-                          "bs_request_actions.#{source_or_target}_package=#{quote(package)})"
-                      end
+      def bs_request_actions_conditions(source_or_target)
+        return unless roles.empty? || roles.include?(source_or_target)
+
+        bs_request_actions_filters = []
+        if project_name.present?
+          bs_request_actions_filters << if subprojects.blank?
+                                          ["bs_request_actions.#{source_or_target}_project = ?", project_name]
+                                        else
+                                          ["bs_request_actions.#{source_or_target}_project like ?", "#{project_name}:%"]
+                                        end
+        end
```

Comment:
```suggestion
      bs_request_actions_filters << if project_name.present? && subprojects.blank?
                                      ["bs_request_actions.#{source_or_target}_project = ?", project_name]
                                    elsif project_name.present?
                                      ["bs_request_actions.#{source_or_target}_project like ?", "#{project_name}:%"]
                                    end
```
I think it's easier to understand this way

Reply:
With your suggestion, if none of the conditions are met, `bs_request_actions_filters` will be `[nil]`. And we don't want that...

Reply:
Yeah, it has to be adapted. But the central idea was to flatten the conditions to avid nesting

Reply:
We also check twice unnecessarily for `project_name.present?`. I still don't see the benefit...

Reply:
Alright, np